File: backend/app/schemas/report.py
"""Report Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID


class ReportCreate(BaseModel):
    title: str
    report_type: str
    document_id: Optional[UUID] = None


class ReportResponse(BaseModel):
    id: UUID
    title: str
    report_type: str
    summary: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ReportDetailResponse(ReportResponse):
    content: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None
    citations: List[Dict[str, Any]] = []
    risk_findings: List[Dict[str, Any]] = []

File: shared/types/report.ts
// Shared Types - Report
export type ReportType = 'risk_analysis' | 'compliance_audit' | 'contract_summary' | 'legal_brief' | 'case_analysis';
export type ReportStatus = 'generating' | 'completed' | 'failed';

export interface Report {
  id: string;
  title: string;
  reportType: ReportType;
  summary?: string;
  status: ReportStatus;
  createdAt: string;
  updatedAt: string;
}

export interface ReportDetail extends Report {
  content?: string;
  metadata?: Record<string, any>;
  citations: Citation[];
  riskFindings: RiskFinding[];
}

export interface RiskFinding {
  severity: 'low' | 'medium' | 'high' | 'critical';
  title: string;
  section: string;
}

import { Citation } from './chat';
export type { Citation };
